OSDOCS-6348: Adding IPFIX exporter option (#63324)
openshift/openshift-docs: modules/network-observability-enriched-flows.adoc

// network_observability/configuring-operator.adoc

:_content-type: PROCEDURE
[id="network-observability-enriched-flows_{context}"]
= Export enriched network flow data

> You can send network flows to Kafka, IPFIX,or both at the same time. Any processor or storage that supports Kafka or IPFIX input, such as Splunk, Elasticsearch, or Fluentd, can consume the enriched network flow data.

Author: IBMSG advises to use commas to separate items in a series of three or more, and use a comma before the conjunction that precedes the final item. In this case, I would need the comma before or.

> You can send network flows to Kafka, IPFIX,or both at the same time. Any processor or storage that supports Kafka or IPFIX input, such as Splunk, Elasticsearch, or Fluentd, can consume the enriched network flow data.

Author: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-commas#commas-between-items-in-a-series

> You can send network flows to Kafka, IPFIX,or both at the same time. Any processor or storage that supports Kafka or IPFIX input, such as Splunk, Elasticsearch, or Fluentd, can consume the enriched network flow data.

Reviewer: oh thanks, I'm learning something here :smile:

.Prerequisites

> * Installed Kafka if choosing to export to Kafka.

Reviewer 2: ```suggestion
* You have installed Kafka, if you choose to export to Kafka.
```

> * You have installed Kafka, if you choose to export to Kafka.

Reviewer: ```suggestion
* Your Kafka or IPFIX collector endpoint(s) are available from Network Observability flowlogs-pipeline pods.
```

Here Kafka or any IPFIX collector may even be outside of the cluster. It just need to be accessible and ready.

Author: Okay, I re-worked this basically just added what you suggest :) TY!

  name: cluster
spec:
  exporters:
  - type: KAFKA                         <3>
      kafka:
        address: "kafka-cluster-kafka-bootstrap.netobserv"
        topic: netobserv-flows-export   <1>
        tls: 
          enable: false                 <2>
  - type: IPFIX                         <3>
      ipfix:
        targetHost: "ipfix-collector.ipfix.svc.cluster.local"
        targetPort: 4739
        transport: tcp or udp           <4>

  
----
<1> The Network Observability Operator exports all flows to the configured Kafka topic.
<2> You can encrypt all communications to and from Kafka with SSL/TLS or mTLS. When enabled, the Kafka CA certificate must be available as a ConfigMap or a Secret, both in the namespace where the `flowlogs-pipeline` processor component is deployed (default: netobserv). It must be referenced with `spec.exporters.tls.caCert`. When using mTLS, client secrets must be available in these namespaces as well (they can be generated for instance using the AMQ Streams User Operator) and referenced with `spec.exporters.tls.userCert`.
<3> You can export flows to IPFIX instead of or in conjunction with exporting flows to Kafka. 

> <4> You have the option to specify transport. The defaults value is `tcp` but you can also specify `udp`.

Reviewer 3: `default` instead of `defaults`

Author: Addressed

        topic: netobserv-flows-export   <1>
        tls: 
          enable: false                 <2>

>      - type: IPFIX                      <3>

Reviewer: ```suggestion
  - type: IPFIX                      <3>
```

Indentation is wrong here ? maybe also on the arguments below

Author: Addressed for all

       ipfix:
         targetHost: "ipfix-collector.ipfix.svc.cluster.local"
         targetPort: 4739
         transport: tcp or udp          <4>

  
----
<1> The Network Observability Operator exports all flows to the configured Kafka topic.
<2> You can encrypt all communications to and from Kafka with SSL/TLS or mTLS. When enabled, the Kafka CA certificate must be available as a ConfigMap or a Secret, both in the namespace where the `flowlogs-pipeline` processor component is deployed (default: netobserv). It must be referenced with `spec.exporters.tls.caCert`. When using mTLS, client secrets must be available in these namespaces as well (they can be generated for instance using the AMQ Streams User Operator) and referenced with `spec.exporters.tls.userCert`.

> <3> You can export flows to IPFIX instead of or in conjunction with exporting flows to Kafka. 

Reviewer: Can we have the note on both `type: KAFKA` & `type: IPFIX` ?

Author: Yes, added

> <4> You have the option to specify trasport. The defaults value is `tcp` but you can also specify `udp`.

Reviewer 3: typo in spelling of transport
